### PR TITLE
fix: ImageData extent and vtkSmartVolumeMapper

### LINKED ISSUE #14: vtkSmartVolumeMapper serializer fails
```py
import pyvista as pv
from pyvista import examples

await pv.set_jupyter_backend('trame')

vol = pv.Wavelet()

p = pv.Plotter()
p.add_volume(vol)
p.show()
```

The local view is empty and the serializer is giving the error

```
ERROR:vtkmodules.web.render_window_serializer:!!!No serializer for vtkSmartVolumeMapper with id 00000002850ad240
```

No matter which volume mapper I use, it seems volume rendered data aren't well supported

### PR DESCRIPTION
**Pull request by banesullivan**
Resolves #14 and other volume/image data issues

### CODE REVIEW

File: trame_vtk/modules/vtk/addon_serializer.py [hunk 1]
```diff
@@ -338,6 +338,33 @@ def cubeAxesSerializer(parent, actor, actorId, context, depth):
...
+    extractRequiredFields(fields, parent, dataset, context, "*")
+
+    return {
+        "parent": render_window_serializer.getReferenceId(parent),
+        "id": datasetId,
+        "type": render_window_serializer.class_name(dataset),
+        "properties": {
+            "spacing": dataset.GetSpacing(),
+            "origin": dataset.GetOrigin(),
+            "dimensions": dataset.GetDimensions(),
```
Comment (banesullivan) [inline trame_vtk/modules/vtk/addon_serializer.py:360]: @jourdain, do I remove this? It's working as expected with these changes, I think...

File: trame_vtk/modules/vtk/addon_serializer.py [hunk 2]
```diff
@@ -338,6 +338,33 @@ def cubeAxesSerializer(parent, actor, actorId, context, depth):
     }
 
 
+def imagedataSerializer(
+    parent, dataset, datasetId, context, depth, requested_fields=["Normals", "TCoords"]
```
[the remark this answers is not in the document]
Comment (jourdain) [reply trame_vtk/modules/vtk/addon_serializer.py:342]: no but we should probably replace `"Normals", "TCoords"` by `vtkGhostType`

File: trame_vtk/modules/vtk/addon_serializer.py (hunk 1 above)
Comment (jourdain) [reply trame_vtk/modules/vtk/addon_serializer.py:360]: yes remove dimensions.

File: trame_vtk/modules/vtk/addon_serializer.py [hunk 3]
```diff
@@ -338,6 +338,88 @@ def cubeAxesSerializer(parent, actor, actorId, context, depth):
...
+            dependencies.append(dataObjectInstance)
+            calls.append(
+                ["setInputData", [render_window_serializer.wrapId(dataObjectId)]]
+            )
+
+    if dataObjectInstance:
+        if hasattr(mapper, "GetImageSampleDistance"):
+            imageSampleDistance = mapper.GetImageSampleDistance()
+        else:
+            imageSampleDistance = 1.0
```
Comment (banesullivan) [inline trame_vtk/modules/vtk/addon_serializer.py:400]: I think this is the best we can do? 1.0 seems to be a reasonable default

Comment (jourdain) [reply trame_vtk/modules/vtk/addon_serializer.py:400]: sure

File: trame_vtk/modules/vtk/addon_serializer.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jourdain) [reply trame_vtk/modules/vtk/addon_serializer.py:342]: I mean we should but for now I think we can ignore it

**Review by jourdain — APPROVED** [review]
LGTM

**Pull request merged**